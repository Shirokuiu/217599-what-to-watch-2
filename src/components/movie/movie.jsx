import React from "react";
import PropTypes from "prop-types";
import {Link} from "react-router-dom";

import VideoPlayer from "../video-player/video-player";

export const Movie = (props) => {
  const {movie, onMovieEnter, onMovieLeave, isPlaying} = props;
  const {previewImage, name, previewVideoLink, id} = movie;

  return <Link
    to={`/movie/${id}/overview`}
    className="small-movie-card catalog__movies-card"
    onMouseEnter={onMovieEnter}
    onMouseLeave={onMovieLeave}
  >
    <div className="small-movie-card__image">
      <VideoPlayer
        src={previewVideoLink}
        poster={previewImage}
        muted={true}
        width={280}
        height={175}
        playerState={isPlaying}
      />
    </div>
    <h3 className="small-movie-card__title">
      <span className="small-movie-card__link">{name}</span>
    </h3>
  </Link>;
};

Movie.propTypes = {
  movie: PropTypes.shape({
    id: PropTypes.number,
    name: PropTypes.string,
    posterImage: PropTypes.string,
    previewImage: PropTypes.string,
    backgroundImage: PropTypes.string,
    backgroundColor: PropTypes.string,
    videoLink: PropTypes.string,
    previewVideoLink: PropTypes.string,
    description: PropTypes.string,
    rating: PropTypes.number,
    scoresCount: PropTypes.number,
    director: PropTypes.string,
    starring: PropTypes.array.string,
    runTime: PropTypes.number,
    genre: PropTypes.string,
    released: PropTypes.number,
    isFavorite: PropTypes.bool,
  }).isRequired,
  onMovieEnter: PropTypes.func,
  onMovieLeave: PropTypes.func,
  isPlaying: PropTypes.bool
};
