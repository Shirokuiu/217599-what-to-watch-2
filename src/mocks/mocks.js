export const films = [
  {
    img: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    title: `Fantastic Beasts`
  },
  {
    img: `img/bohemian-rhapsody.jpg`,
    title: `Bohemian Rhapsody`
  },
  {
    img: `img/macbeth.jpg`,
    title: `Macbeth`
  },
  {
    img: `img/aviator.jpg`,
    title: `Aviator`
  },
  {
    img: `img/we-need-to-talk-about-kevin.jpg`,
    title: `We need to talk about Kevin`
  }
];
