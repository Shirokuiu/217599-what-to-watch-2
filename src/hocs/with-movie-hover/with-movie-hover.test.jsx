import React from "react";
import renderer from "react-test-renderer";

import {Movie} from "../../components/movie/movie";

import {films} from "../../mocks/mocks";

it(`Is movie render`, () => {
  const tree = renderer
    .create(<Movie
      film={films[0]}
      id={0}
      key={0}
      onMovieClick={jest.fn()}
      onMovieEnter={jest.fn()}
      onMovieLeave={jest.fn()}
    />).toJSON();

  expect(tree).toMatchSnapshot();
});
