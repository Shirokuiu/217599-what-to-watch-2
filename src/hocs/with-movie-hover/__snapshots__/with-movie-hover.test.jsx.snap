// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Is movie render 1`] = `
<a
  className="small-movie-card catalog__movies-card"
  href="/movie/1/overview"
  onClick={[Function]}
  onMouseEnter={[MockFunction]}
  onMouseLeave={[MockFunction]}
>
  <div
    className="small-movie-card__image"
  >
    <video
      height={175}
      muted={true}
      poster="img/the-grand-budapest-hotel.jpg"
      src="https://some-link"
      width={280}
    />
  </div>
  <h3
    className="small-movie-card__title"
  >
    <span
      className="small-movie-card__link"
    >
      The Grand Budapest Hotel
    </span>
  </h3>
</a>
`;
